examples/animation/double_pendulum.py

```diff
@@ -71,22 +73,30 @@ def derivs(state, t):
 y2 = -L2*cos(y[:, 2]) + y1
 
 fig = plt.figure(figsize=(5, 4))
-ax = fig.add_subplot(autoscale_on=False, xlim=(-2, 2), ylim=(-2, 1))
+ax = fig.add_subplot(autoscale_on=False,
+                     xlim=(-(L1+L2), (L1+L2)),
+                     ylim=(-(L1+L2), 1.))
 ax.set_aspect('equal')
 ax.grid()
 
 line, = ax.plot([], [], 'o-', lw=2)
+trace, = ax.plot([], [], ',-', lw=1)
 time_template = 'time = %.1fs'
 time_text = ax.text(0.05, 0.9, '', transform=ax.transAxes)
+history_x, history_y = deque(maxlen=history_len), deque(maxlen=history_len)
 
 
 def animate(i):
     thisx = [0, x1[i], x2[i]]
     thisy = [0, y1[i], y2[i]]
 
+    history_x.appendleft(thisx[2])
+    history_y.appendleft(thisy[2])
+
     line.set_data(thisx, thisy)
+    trace.set_data(history_x, history_y)
```

Comment:
Because the animation repeats, history should be deleted at `i == 0`. Otherwise you still see the trace remainder of the previous run.

Reply:
Good catch! I have fixed it.